Assets/LeapMotionModules/InteractionEngine/Scripts/InteractionC/InteractionC.cs

```diff
@@ -398,6 +404,34 @@ public class InteractionC {
       return rs;
     }
 
+    /*** Get Velocities ***/
+    [DllImport(DLL_NAME, EntryPoint = "LeapIEGetVelocities")]
+    private static extern eLeapIERS LeapIEGetVelocities(ref LEAP_IE_SCENE scene,
+                                                        out UInt32 nVelocities,
+                                                        out IntPtr ppVelocitiesBuffer);
+
+    public static eLeapIERS GetVelocities(ref LEAP_IE_SCENE scene, out LEAP_IE_VELOCITY[] velocities)
+    {
+      Logger.Log("Get Velocities", LogLevel.AllCalls);
+
+      UInt32 nVelocities;
+      IntPtr ppVelocitiesBuffer;
+      var rs = LeapIEGetVelocities(ref scene, out nVelocities, out ppVelocitiesBuffer);
+      Logger.HandleReturnStatus(rs);
+      if (rs != eLeapIERS.eLeapIERS_Success || nVelocities == 0)
+      {
+        velocities = null;
+        return rs;
+      }
+
+      velocities = new LEAP_IE_VELOCITY[nVelocities];
+      for (int i = 0; i < nVelocities; i++)
+      {
+        velocities[i] = StructMarshal<LEAP_IE_VELOCITY>.ArrayElementToStruct(ppVelocitiesBuffer, i);
```

Comment:
is ppVelocitiesBuffer a pointer to an array of pointers?  If so, I think you would want StructMarshal<IntPtr> instead